torch_xla/test/metrics_test_wrapper.py

```diff
@@ -0,0 +1,131 @@
+"""Wrap PyTorch XLA training jobs with extra logic around XLA metrics.
+
+It can be used to save a historical record of metrics and/or check metrics
+against a golden set of metrics to search for performance regressions.
+
+This script expects a certain directory structure to setup tests and to hold
+metrics results. In the outermost directory, there should be a base config file
+detailed below. Within that directory, there should be a subdirectory for each
+test that contains a metrics_history subdirectory and (optionally) a config file
+to override fields of the base config. In other words, it should look like this:
+
+    root/
+      base_config.json
+      mnist/
+        config_overrides.json
+        metrics_history/
+
+TODO(zcain): Design and document the config file.
+
+Example usage:
+python metrics_test_wrapper.py --root="gs://model_metrics" \
+    --test_folder_name="mnist" -- python test/test_train_mnist.py --num_epochs=1
+"""
+import argparse
+import datetime
+import glob
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+
+import torch_xla.utils.gcsfs as gcsfs
+
+
+_CLOUD_STORAGE_PREFIX = 'gs://'
+_METRICS_HISTORY_DIR_NAME = 'metrics_history'
+_XLA_METRICS_FILE = 'XLA_METRICS_FILE'
+
+
+def _find_correct_metrics_file(base_metrics_file):
+  # Distributed training jobs write 1 file per core. If XLA_METRICS_FILE is set
+  # to 'base_metrics_file', then the distributed output files have the format
+  # 'base_metrics_file.0', 'base_metrics_file.1', etc.
+  metrics_files = glob.glob('{}*'.format(base_metrics_file))
+  if len(metrics_files) == 1:
+      # Non-distributed case: the correct file is simply 'base_metrics_file'.
+    file_to_read = base_metrics_file
+  else:
+    # Choose the file of the first core. When sorted alphabetically, the first
+    # element will be 'base_metrics_file' which is empty in the distributed
+    # case. The second element will be 'base_metrics_file.0'.
+    file_to_read = sorted(metrics_files)[1]
+  return file_to_read
+
+
+def _run_subprocess(cmd):
+  # Use _XLA_METRICS_FILE to pass the metrics report from the subprocess
+  # to this process.
+  tmp_dir = tempfile.mkdtemp()
+  _, tmp_path = tempfile.mkstemp(dir=tmp_dir)
+
+  # Run the PyTorch XLA script.
+  os.environ[_XLA_METRICS_FILE] = tmp_path
+  sp_return_code = subprocess.call(cmd)
+
+  with open(_find_correct_metrics_file(tmp_path), 'r') as tmp_metrics:
+    metrics = tmp_metrics.read()
+
+  # Cleanup.
+  shutil.rmtree(tmp_dir)
+  del os.environ[_XLA_METRICS_FILE]
+
+  # Use only the metrics from the last step.
+  metrics = metrics[metrics.rfind('[MetricsData;'):]
+  return metrics, sp_return_code
+
+
+def _write_to_disk(output_string, output_filename):
```

Comment:
How do we handle TTL to expire old files?
Manually?

Reply:
The plan was to avoid any kind of TTL for now since we want to hold onto our historical data and the storage cost is low. If we run into issues and need TTL, I can set it on the cloud storage bucket that holds the metrics history. No need for logic in this file

Reply:
We will have to glob there, so having 1000s of file might be an issue.
But we could come back later to it, eventually.

Reply:
I see. We could maybe have a cronjob to move files from a `recent_history` directory to a `history` directory such that `recent_history`. I could probably have this as an automated workflow in the cloud project. But like you said, we can revisit when we have more files